toolchain/install/busybox_info.h

```diff
@@ -35,8 +35,19 @@ struct BusyboxInfo {
 // If unable to locate a plausible busybox binary, returns an error instead.
 inline auto GetBusyboxInfo(llvm::StringRef argv0) -> ErrorOr<BusyboxInfo> {
   // Check for an override of `argv[0]` from the environment and apply it.
+  std::string argv0_override_storage;
   if (const char* argv0_override = getenv(Argv0OverrideEnv)) {
-    argv0 = argv0_override;
+    // Capture the override into local storage here so we can clear it from the
+    // environment.
+    argv0_override_storage = argv0_override;
+    argv0 = argv0_override_storage;
+
+    // Unset the override environment variable so that subsequent subprocessing
+    // back into the busybox doesn't pick it up and uses the subprocesses's
+    // argv[0]. This is essential to, for example, allow the `clang` driver to
+    // subprocess back into a `clang` invocation with an internal CC1
```

Comment:
What you mean by a "`clang` invocation"? Perhaps:

```suggestion
    // subprocess back into a `clang` busyboxed binary with an internal CC1
```